js/core/index.js

```diff
@@ -43,3 +57,44 @@ function Runtime() {
 }
 
 global.runtime = module.exports = new Runtime();
+
+runtime.tty.stdout = new runtime.tty.StdOut();
+runtime.tty.stdin = new runtime.tty.StdIn();
+initio(runtime);
+
+runtime.stdio = {
+  stdout: runtime.tty.stdout,
+  stdin: runtime.tty.stdin,
+  StdIn: runtime.tty.StdIn,
+  StdOut: runtime.tty.StdOut,
+  createOut: function(funcs) {
+    if (funcs) {
+      funcs.onwrite = funcs.onwrite || runtime.stdio.stdout.onwrite;
+      funcs.onsetcolor = funcs.onsetcolor || runtime.stdio.stdout.onsetcolor;
+      funcs.onsetbackgroundcolor = funcs.onsetbackgroundcolor || runtime.stdio.stdout.onsetbackgroundcolor;
```

Comment:
I think we should not default those functions to stdout, it could be very unexpected. Let's just assign null by default that does nothing.